sky/backends/cloud_vm_ray_backend.py

```diff
@@ -3946,7 +3956,9 @@ def _execute_task_one_node(self, handle: CloudVmRayResourceHandle,
                                 codegen.build(),
```

Comment:
How about
- `codegen.build() + ' && ' + self._maybe_set_spot_task_to_pending_code(...)`?
- add a comment on why inside the generated code, we call spot_state.set_pend... only after the `codegen` command has been run?

Reply:
I changed the implementation to have the spot_task passed to the underlying `exec_code_on_head`.

- We cannot directly append the code to the `codegen.build()`, as `codegen.build()` is a python program and will be submitted with `ray job submit`. The setting of the spot status should not be within the ray job, as the `ray job submit` will now only be called after the skypilot job is removed from the pending table, i.e. a new empty slot for the resources becomes available, whcih will make the spot task not appear in the spot queue until another controller process finishes and free up a slot on the controller VM.

- I added a comment in the place we add the spot status setting code. PTAL : )